File: Raven.Client.Java/src/main/java/net/ravendb/client/IDocumentSessionImpl.java
package net.ravendb.client;

import net.ravendb.abstractions.basic.Lazy;
import net.ravendb.abstractions.basic.Tuple;
import net.ravendb.abstractions.closure.Action1;
import net.ravendb.client.document.DocumentConvention;
import net.ravendb.client.document.batches.IEagerSessionOperations;

/**
 * Interface for document session which holds the internal operations
 */
public interface IDocumentSessionImpl extends IDocumentSession, IEagerSessionOperations {

  public DocumentConvention getConventions();

  public <T> T[] loadInternal(Class<T> clazz, String[] ids);

  public <T> T[] loadInternal(Class<T> clazz, String[] ids, Tuple<String, Class<?>>[] includes);

  <T> Lazy<T[]> lazyLoadInternal(Class<T> clazz, String[] ids, Tuple<String, Class<?>>[] includes, Action1<T[]> onEval);
}

File: src/Raven.NewClient/Documents/IDocumentSessionImpl.cs
//-----------------------------------------------------------------------
// <copyright file="IDocumentSessionImpl.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
using System;
using System.Collections.Generic;
using Raven.NewClient.Client.Document;
using Raven.NewClient.Client.Document.Batches;
using Raven.NewClient.Json.Linq;

namespace Raven.NewClient.Client.Documents
{
    /// <summary>
    /// Interface for document session which holds the internal operations
    /// </summary>
    internal interface IDocumentSessionImpl : IDocumentSession, ILazySessionOperations, IEagerSessionOperations
    {
        DocumentConvention Conventions { get; }

        T[] LoadInternal<T>(string[] ids);
        T[] LoadInternal<T>(string[] ids, KeyValuePair<string, Type>[] includes);
        T[] LoadInternal<T>(string[] ids, string transformer, Dictionary<string, object> transformerParameters = null);
        T[] LoadInternal<T>(string[] ids, KeyValuePair<string, Type>[] includes, string transformer, Dictionary<string, object> transformerParameters = null);
        Lazy<T[]> LazyLoadInternal<T>(string[] ids, KeyValuePair<string, Type>[] includes, Action<T[]> onEval);
    }
}
